File: Selenium/Activity8.py
from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By

with webdriver.Firefox() as driver:
    actions = ActionChains(driver)
    driver.get("https://training-support.net/webelements/mouse-events")
    print("Page title is: ", driver.title)

    cargoLock = driver.find_element(By.XPATH, "//h1[text()='Cargo.lock']")
    cargoToml = driver.find_element(By.XPATH, "//h1[text()='Cargo.toml']")
    srcButton = driver.find_element(By.XPATH, "//h1[text()='src']")
    targetButton = driver.find_element(By.XPATH, "//h1[text()='target']")

    actions.click(cargoLock).pause(1).move_to_element(cargoToml).click(cargoToml).perform()
    actionMessage = driver.find_element(By.ID, "result").text
    print("Confirmation message", actionMessage)

    actions.double_click(srcButton).context_click(targetButton).perform()
    actions.click(driver.find_element(By.XPATH, ("//div[@id='menu']/div/ul/li[1]"))).pause(5).perform()
    actionMessage1 = driver.find_element(By.ID, "result").text
    print("Confirmation message", actionMessage1)
    driver.quit()

File: Selenium/Activity8.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity8 {

	public static void main(String[] args) {
		
		WebDriver driver = new FirefoxDriver();
        Actions builder = new Actions(driver);
        driver.get("https://training-support.net/webelements/mouse-events");
        System.out.println("Page title: " + driver.getTitle());

        WebElement cargoLock = driver.findElement(By.xpath("//h1[text()='Cargo.lock']"));
        WebElement cargoToml = driver.findElement(By.xpath("//h1[text()='Cargo.toml']"));
        WebElement srcButton = driver.findElement(By.xpath("//h1[text()='src']"));
        WebElement targetButton = driver.findElement(By.xpath("//h1[text()='target']"));

        
        builder.click(cargoLock).moveToElement(cargoToml).click(cargoToml).build().perform();
        String actionMessage = driver.findElement(By.id("result")).getText();
        System.out.println("Confirmation message: "+actionMessage);

        builder.doubleClick(srcButton).contextClick(targetButton).build().perform();
        builder.click(driver.findElement(By.xpath("//div[@id='menu']/div/ul/li[1]"))).pause(3000).build().perform();
        actionMessage = driver.findElement(By.id("result")).getText();
        System.out.println("Confirmation message: "+actionMessage);
        driver.quit();

	}

}
